sdk/messaging/azservicebus/internal/amqpLinks.go

```diff
@@ -46,40 +51,39 @@ type AMQPLinks interface {
 	ClosedPermanently() bool
 }
 
-// amqpLinks manages the set of AMQP links (and detritus) typically needed to work
+// AMQPLinksImpl manages the set of AMQP links (and detritus) typically needed to work
 //  within Service Bus:
 // - An *goamqp.Sender or *goamqp.Receiver AMQP link (could also be 'both' if needed)
 // - A `$management` link
 // - an *goamqp.Session
 //
 // State management can be done through Recover (close and reopen), Close (close permanently, return failures)
-// and Get() (retrieve latest version of all amqpLinks, or create if needed).
-type amqpLinks struct {
+// and Get() (retrieve latest version of all AMQPLinksImpl, or create if needed).
+type AMQPLinksImpl struct {
+	// NOTE: values need to be 64-bit aligned. Simplest way to make sure this happens
+	// is just to make it the first value in the struct
+	// See:
+	//   Godoc: https://pkg.go.dev/sync/atomic#pkg-note-BUG
+	//   PR: https://github.com/Azure/azure-sdk-for-go/pull/16847
+	id LinkID
+
 	entityPath     string
 	managementPath string
 	audience       string
 	createLink     CreateLinkFunc
-	baseRetrier    Retrier
 
 	mu sync.RWMutex
 
-	// mgmt lets you interact with the $management link for your entity.
-	mgmt MgmtClient
+	// RPCLink lets you interact with the $management link for your entity.
+	RPCLink RPCLink
 
 	// the AMQP session for either the 'sender' or 'receiver' link
 	session AMQPSessionCloser
 
 	// these are populated by your `createLinkFunc` when you construct
 	// the amqpLinks
-	sender   AMQPSenderCloser
-	receiver AMQPReceiverCloser
-
-	// last connection revision seen by this links instance.
-	clientRevision uint64
```

Comment:
These two "revisions" (which are basically just IDs) are in the LinkID struct now.